Client: Update Section Header component to Typescript file

kirahsapong opened the pull request:
Why: To reap the benefits of TypeScript (such as added safety and automation) for this file. Also, a good way to practice creating a PR for Jetpack.

`cardBadge` prop and its use in `dash-item` should be removed in a second PR. This prop was removed in [a previous PR](https://github.com/Automattic/jetpack/pull/9732). 

`SectionHeader` component should make use of semantic `h` level tags in a follow-up PR for improved a11y. 

#### Changes proposed in this Pull Request:

* Convert `SectionHeader` component from Javascript to Typescript file
* Add `cardBadge` optional prop back into component for backwards compatibility

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
1. Pull up Jetpack Settings screen at `/wp-admin/admin.php?page=jetpack#/settings`
2. Section Headings should be visible per the screenshot below:
<img width="1026" alt="Screen Shot 2022-10-05 at 12 40 26 PM" src="https://user-images.githubusercontent.com/102400653/194115072-ad37a3c4-5741-4095-823b-4da73362e5c5.png">

Commit:
[not verified] Client: Update Section Header component to Typescript file

Why: To reap the benefits of TypeScript (such as added safety and automation) for this file.
Also, a good way to practice creating a PR for Jetpack.

jeyip commented:
For the typed section header file, I'm seeing a "can't find module" error. Did you / do you happen to see the same thing?

<img width="876" alt="Screen Shot 2022-10-06 at 12 19 14 AM" src="https://user-images.githubusercontent.com/5414230/194243012-a1265f97-f807-4cc8-97cb-8154134acef1.png">

jeyip commented:
## Testing

### Requirements
- [x] Section Headings should be visible

### Browsers
- [x] Chrome

jeyip commented on projects/plugins/jetpack/_inc/client/components/section-header/index.tsx:
```diff
@@ -0,0 +1,26 @@
+import './style.scss';
+
+interface SectionHeaderProps {
+	label?: string;
```
`NB`: Should we allow this to be optional? If so, maybe it should have a default value of an empty string?

kirahsapong replied:
@jeyip Good catch. Where it's been implemented (`DashItem`, `PluginDashItem`, and `SettingsCard`) `label` is always passed, which makes sense here given how its used. Have updated the prop to be required. Thank you!

jeyip replied:
Thank you! [ba7f535](https://github.com/Automattic/jetpack/pull/26644/commits/ba7f535f684731b4bf5b491b4274301907aa37d9) looks good to me 👍

jeyip reviewed (commented):
Changes generally look good 🎉  -- just curious about https://github.com/Automattic/jetpack/pull/26644#issuecomment-1269476162

+1 from me when the comment is resolved

Commit:
[not verified] Set typed label property as required in SectionHeaderProps interface

The label property is passed everywhere this component is implemented, which makes sense given the component displays the value of the label property as the SectionHeader content

kirahsapong commented:
> For the typed section header file, I'm seeing a "can't find module" error. Did you / do you happen to see the same thing?

No, everything looks good on my end! Have you tried building on your end?

jeyip commented:
> No, everything looks good on my end! Have you tried building on your end?

I'll give it a shot. Either way, it must be something on my end. I vaguely remember seeing something similar a while ago on one of my PRs where typescript annotations weren't being updated in build files.

sharpirate reviewed (approved):
✅ This looks good on my end 

<img width="1147" alt="image" src="https://user-images.githubusercontent.com/20927667/195020436-46b54989-322b-4950-a0e9-7ba4fbe3cdf0.png">

obenland committed:
Merge branch 'trunk' into update/section-header-to-typescript
